tests/trainer/connectors/test_data_connector.py

```diff
@@ -11,20 +11,388 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from contextlib import redirect_stderr
+from io import StringIO
+from re import escape
 from unittest.mock import Mock
 
 import pytest
-from torch.utils.data import DataLoader
+from torch.utils.data import BatchSampler, DataLoader, DistributedSampler, Sampler, SequentialSampler
 
 from pytorch_lightning import Trainer
+from pytorch_lightning.strategies import DDPSpawnStrategy
 from pytorch_lightning.trainer.connectors.data_connector import _DataHookSelector, _DataLoaderSource, warning_cache
-from pytorch_lightning.trainer.states import TrainerFn
+from pytorch_lightning.trainer.states import RunningStage, TrainerFn
+from pytorch_lightning.trainer.supporters import CombinedLoader
+from pytorch_lightning.utilities.data import _update_dataloader
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from pytorch_lightning.utilities.warnings import PossibleUserWarning
-from tests.helpers import BoringDataModule, BoringModel
-from tests.helpers.boring_model import RandomDataset
+from tests.helpers.boring_model import BoringDataModule, BoringModel, RandomDataset
+from tests.helpers.runif import RunIf
 from tests.helpers.utils import no_warning_call
 
 
+@RunIf(skip_windows=True)
+@pytest.mark.parametrize("mode", (1, 2))
+def test_replace_distributed_sampler(tmpdir, mode):
```

Comment:
Can you please point out which tests are new? Or push refactor to a new PR. It makes a reviewers life much easier :)

Reply:
mentioned here: https://github.com/PyTorchLightning/pytorch-lightning/pull/12653#issuecomment-1091595756

Reply:
added to description.

Reply:
just came back here to do that... ahaaha! better :) thanks